src/main/java/org/mitre/synthea/world/agents/Person.java

```diff
@@ -399,7 +407,15 @@ public Double getVitalSign(VitalSign vitalSign, long time) {
       default:
         decimalPlaces = 2;
     }
-    return BigDecimal.valueOf(value).setScale(decimalPlaces, RoundingMode.HALF_UP).doubleValue();
+    Double retVal = value;
+    try {
+      retVal = BigDecimal.valueOf(value)
+              .setScale(decimalPlaces, RoundingMode.HALF_UP)
+              .doubleValue();
+    } catch (NumberFormatException e) {
```

Comment:
I have mixed feelings about this. If something is infinity or not a number, the code still works, but we swallow the formatting error. I don't think any vital sign ever should be infinity or NaN, so it seems like we should find some way of warning that something has gone wrong.

I say this as someone who was guilty of a bug in a module that was causing BMI to be NaN.

Reply:
Rather than swallowing/propagating the issue here, maybe we should add explicit checks elsewhere that will throw an exception on NaN or infinity?

Reply:
Yeah. A good place for this case would be in `setVitalSign`. There are other places that would probably be good to have sanity checks, but that's a discussion for a separate thread.